### DISCUSSION TITLE
Mise GUI/TUI/IDE extensions

### DISCUSSION THREAD

**Discussion by hverlin** [jdx/mise#2898] [Ideas]
> GUI/TUI - While we're all big CLI fans, it still would be great to better visualize what tools are available, what your configuration is, and other things via some kind of UI.

https://mise.jdx.dev/project-roadmap.html

- Would you see `mise` having a built-in UI (using [iced](https://github.com/iced-rs/iced) for example)? Or TUI (example [ratatui](https://github.com/ratatui/ratatui))
- It could be an external tool (probably this would mean that most commands should offer a `--json` output for easier parsing)
- Another option would be to have some UI by using IDE plugins? (I believe this would also require having a `--json` option for most commands)

I think it could also be useful to be able to manage tasks and environment variables.

**Comment by nickchomey** [comment]
I'd love to see a nice TUI

**Comment by hverlin** [comment]
Started to add a few `mise` UI features in https://github.com/hverlin/mise-vscode and https://github.com/134130/intellij-mise
